src/appleseed/renderer/modeling/aov/cryptomatteaov.cpp

```diff
@@ -527,8 +533,14 @@ namespace
                             pixel_values.push_back(coverage);
                         }
 
+                        // Set the remaining channels of the pixel to black.
+                        // This is determined by taking 2 channels per ranked vector plus
+                        // 3 channels for the preview image and subtracting that from the
+                        // total number of AOV channels.
                         const size_t num_channels = (m_num_layers * 2) + 3;
-                        for (size_t i = ranked_vector.size(); i < num_channels; ++i)
+                        const size_t filled_pixels = ranked_vector_size * 2 + 3;
```

Comment:
I guess what would be clearer and simpler is to get rid of `ranked_vector_size` and use ` ranked_vector.size() - ranked_vector_start` here.

Reply:
I think `filled_pixels` isn't quite right, this isn't about pixels but about channels, right?

Reply:
Good point.  Got my terms crossed

Reply:
I see what you're saying with regards to the 'ranked_vector_size' comment.  I'll modify